OpenBazaar/openbazaar-desktop
# Tor

This change-set wires in Tor controls to the client.

closes #315 

To test:
=====
1. Ensure Tor will run on your machine without authentication
a. Download the Tor browser.
b. Start up the Tor browser.
c. Attempt to start your server with the --tor flag. If it works, you'll see this in the terminal (`[Execute] [NOTICE] Using Tor exclusively`) and you're free to proceed to step 2. If it doesn't work, you'll likely see some type of auth error in the terminal. In that case, please ping @rmisio for further debugging.

The following steps will allow your client to masquerade as the bundled app and test Tor scenarios specific to the bundled (installer) version of the app.
 
2. Ensure no OB servers are running.
3. Start-up the client and run `localStorage.clear()` in the console.
4. Shut-down the client.
5. Download the [latest Server release](https://github.com/OpenBazaar/openbazaar-go/releases/) for your OS. Rename it to `openbazaard` on Linux or Mac, `openbazaard.exe` on Windows. Ensure the file is exectuable (`chmod +x on Mac and Linux`)
6. Edit main.js with the path to the executable in the previous step. Find the first line, comment it out and add the second line.

```
// const serverPath = `${__dirname}${path.sep}..${path.sep}openbazaar-go${path.sep}`;
const serverPath = '/path/to/executable/'; // do not include filename, do include trailing `/`
```

4. Start up the client. It will create a server configuration and attempt to connect to the built-in server that you downloaded.

If you are able to set-up a remote server, it would be a really good test to toggle your connection between a remote server and the built-in server and ensure that the connection is properly in Tor mode when you configure it (check 'use Tor') to be.
